### PR TITLE
Upgrade

### LINKED ISSUE #44: Handle upgrades
We need a nice way to handle upgrades.

### PR DESCRIPTION
**Pull request by carolynvs**
Fixes #44. On mac/linux it updates the running binary. On windows it downloads the binary and the wrapper script copies it.

`dvm upgrade` installs the latest version
`dvm upgrade --check` checks for a new version but does not make any changes.
`dvm upgrade --version <version>` installs the specified version.

### CODE REVIEW

File: dvm-helper/dvm-helper.amd64.go [hunk 1]
```diff
@@ -3,3 +3,4 @@
 package main
 
 const dockerArch string = "x86_64"
+const dvmArch string = "x86_64"
```
Comment (rgbkrk) [inline dvm-helper/dvm-helper.amd64.go]: seems like you might as well call this `arch` across these.

Comment (rgbkrk) [inline dvm-helper/dvm-helper.amd64.go]: or even `unameArch`

Comment (carolynvs) [inline dvm-helper/dvm-helper.amd64.go]: At one point they were different but after our changes from Friday, they are the same again. I'll go with `unameArch`.

Comment (rgbkrk) [inline dvm-helper/dvm-helper.amd64.go]: Ah, cool.

**Comment by rgbkrk** [comment]
After this one, let's make a release since I just merged #62.

File: dvm-helper/dvm-helper.go [hunk 2]
```diff
@@ -274,39 +254,9 @@ func install(version string) {
 
 	writeInfo("Installing %s...", version)
 
-	// Download to a temporary location
 	url := buildDownloadURL(version)
-	tmpPath := filepath.Join(getDvmDir(), ".tmp/docker", version, getBinaryName())
```
Comment (rgbkrk) [inline dvm-helper/dvm-helper.go]: The Docker URLs are not 1:1 compatible with ours.

File: dvm-helper/dvm-helper.go [hunk 3]
```diff
@@ -320,7 +270,7 @@ func buildDownloadURL(version string) string {
 		version = "latest"
 	}
 
-	return fmt.Sprintf("%s/%s/%s/docker-%s%s", mirrorURL, dockerOS, dockerArch, version, binaryFileExt)
```
Comment (rgbkrk) [inline dvm-helper/dvm-helper.go]: Here we go.

**Pull request merged**